### ISSUE TITLE
Support for activation literals

### ISSUE THREAD

**Issue by rachitnigam** [elliottt/easy-smt#20]
Not sure if this is goal for this library but it might be useful to implement [activation literals](https://docs.rs/rsmt2/latest/rsmt2/actlit/index.html) in the same way that the `rsmt2` library does. In practice, they are a much more efficient alternative to `push` and `pop` since the latter force SMT solvers to reset a bunch of internal state and disable incremental solving which can be useful for subsequent queries.

**Comment by elliottt** [comment]
That's a neat idea! I'm not sure it's an explicit goal for the library -- originally it started as a rust implementation of the haskell simple-smt library, whose goal was to provide a uniform api for the smt-lib interface that many solvers provide. I think that supporting activation literals is the sort of thing that could be built on top of easy-smt, perhaps we could include a module that would help manage that state for those that would like to use them?

**Comment by rachitnigam** [comment]
Yup! That’s exactly how rsmt2 does it as well. I have a use case for it so might try implementing it. One nice thing in this library is since everything is interned, there might be a cleaner, simpler abstraction to support path dependent/activation literal-based asserts

**Comment by rachitnigam** [comment]
Normally solvers need to reset lots of state and disable various tactics that are not compatible with "incremental solving" (which is what push and pop require).
